File: unit-3/sprint-2/day-1/assignments/Play Some Music/app/src/main/java/com/example/playsomemusic/MainActivity.java
package com.example.playsomemusic;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button btnstart,btnstop;
    private MediaPlayer mediaPlayer;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mediaPlayer = MediaPlayer.create(this,R.raw.songs);
        initiviews();
    }

    private void initiviews() {
        btnstart = findViewById(R.id.startMusic);
        btnstop = findViewById(R.id.StopMusic);
        Intent intent = new Intent(this,MusicService.class);
        btnstart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startService(intent);
            }
        });

        btnstop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                stopService(intent);
            }
        });
    }
}

File: unit-4/sprint-2/day-2/assignments/Money Manager Assignemnt/app/src/main/java/com/example/moneymanager/MainActivity.kt
package com.example.moneymanager

import android.annotation.SuppressLint
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_add.*
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.item_layout_row.*

class MainActivity : AppCompatActivity(), OnitemClickListener {
    lateinit var moneyMangerAdaptor: MoneyMangerAdaptor
    private val tasksList = mutableListOf<Money>()

    private val dbhandler = database(this)

    var amountincome = 0.0
    var amountexpenses = 0.0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        btnadd.setOnClickListener {
            val intent = Intent(this, AddActivity::class.java)
            startActivity(intent)

            updateUi()
        }




        moneyMangerAdaptor = MoneyMangerAdaptor(this, tasksList, this)
        recyclerview.layoutManager = LinearLayoutManager(this)
        recyclerview.adapter = moneyMangerAdaptor
        Income()
        Expenses()

         amountincome = dbhandler.getIncomeExpneses()
         amountexpenses = dbhandler.getExpenseExpneses()
        if (amountincome != null && amountexpenses != null) {
            tvBalanceMoney.text = (amountincome - amountexpenses).toString()
        }


    }


    fun Income() {

        tvIncomeMoney.text = dbhandler?.getIncomeExpneses().toString()
        updateUi()

    }

    fun Expenses() {

        tvExpensesMoney.text = dbhandler?.getExpenseExpneses().toString()
        updateUi()
    }

    override fun OnEdit(money: Money) {

        val intent = Intent(this, AddActivity::class.java)
        startActivity(intent)

        var titleupdate = intent.getStringExtra("updatetitle")
        var amountupdate = intent.getStringExtra("updateamount")

        if (titleupdate != null) {
            money.mTvDescription = titleupdate
        }

        money.mTvDescription = tvIncomeDesc.toString()


        dbhandler.editTask(money)

        updateUi()
    }


    override fun onDelete(money: Money) {

        dbhandler.deleteTask(money)
        updateUi()
    }

    @SuppressLint("NotifyDataSetChanged")
    fun updateUi() {
        val latestTask = dbhandler.getAllTasks()
        tasksList.clear() //if we won't do this then it will create a duplicate values
        tasksList.addAll(latestTask)
        moneyMangerAdaptor.notifyDataSetChanged()
    }
}



